packages/foam-vscode/src/dated-notes.ts

```diff
@@ -0,0 +1,101 @@
+import {
+  Selection,
+  Uri,
+  window,
+  workspace,
+  WorkspaceConfiguration
+} from "vscode";
+import { dirname, join } from "path";
+import dateFormat from "dateformat";
+import * as fs from "fs";
+import { docConfig } from "./utils";
+
+async function openDailyNoteFor(date?: Date) {
+  const foamConfiguration = workspace.getConfiguration("foam");
+  const currentDate = date !== undefined ? date : new Date();
+
+  const dailyNotePath = getDailyNotePath(foamConfiguration, currentDate);
+
+  const isNew = await createDailyNoteIfNotExists(
+    foamConfiguration,
+    dailyNotePath,
+    currentDate
+  );
+  await focusDailyNote(dailyNotePath, isNew);
+}
+function getDailyNotePath(configuration: WorkspaceConfiguration, date: Date) {
+  const rootDirectory = workspace.workspaceFolders[0].uri.fsPath;
+  const dailyNoteDirectory: string =
+    configuration.get("openDailyNote.directory") ?? ".";
+  const dailyNoteFilename = getDailyNoteFileName(configuration, date);
+
+  return join(rootDirectory, dailyNoteDirectory, dailyNoteFilename);
+}
+
+function getDailyNoteFileName(
+  configuration: WorkspaceConfiguration,
+  date: Date
+): string {
+  const filenameFormat: string = configuration.get(
+    "openDailyNote.filenameFormat"
+  );
+  const fileExtension: string = configuration.get(
+    "openDailyNote.fileExtension"
+  );
+
+  return `${dateFormat(date, filenameFormat, false)}.${fileExtension}`;
+}
+
+async function createDailyNoteIfNotExists(
+  configuration: WorkspaceConfiguration,
+  dailyNotePath: string,
+  currentDate: Date
+) {
+  if (await pathExists(dailyNotePath)) {
```

Comment:
TIL you can await within the condition of an `if` statement 🤯

Reply:
I don't think this was actually me...this looks like some moved code (daily notes were refactored to share code) So TIL too!!! 😅

Reply:
It was @francishamel in the PR - https://github.com/foambubble/foam/pull/79/files#diff-c620ef40775aeffc01b5a8468974b524e8f1ce4dfcc842f73ffc0dbf34e53227R60